Source/Core/Common/x64Emitter.h

```diff
@@ -1122,6 +1129,18 @@ class XEmitter
     ABI_CallFunction(func);
   }
 
+  // Pass a pointer and register as a parameter.
+  template <typename FunctionPointer>
+  void ABI_CallFunctionPR(FunctionPointer func, const void* ptr, X64Reg reg1)
+  {
+    MOV(64, R(ABI_PARAM1), Imm64(reinterpret_cast<u64>(ptr)));
+
+    if (reg1 != ABI_PARAM2)
+      MOV(32, R(ABI_PARAM2), R(reg1));
```

Comment:
What happens if `reg1` == `ABI_PARAM1`? This would overwrite the data in reg1 by the pointer before it gets copied into parameter 2. Can that happen?